homeassistant/components/sonarr/entity.py

```diff
@@ -1,41 +1,34 @@
 """Base Entity for Sonarr."""
-from aiopyarr import SystemStatus
-from aiopyarr.models.host_configuration import PyArrHostConfiguration
-from aiopyarr.sonarr_client import SonarrClient
-
 from homeassistant.helpers.device_registry import DeviceEntryType
-from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import DOMAIN
+from .coordinator import SonarrDataUpdateCoordinator
 
 
-class SonarrEntity(Entity):
+class SonarrEntity(CoordinatorEntity[SonarrDataUpdateCoordinator]):
```

Comment:
`SonarrDataUpdateCoordinator` is a Generic.
```suggestion
class SonarrEntity(CoordinatorEntity[SonarrDataUpdateCoordinator[SonarDataT]):
```